### PR TITLE
AP_Arming: Add servo voltage check, paramertise the Vcc check

### PR DESCRIPTION
**Pull request by WickedShell**
Needed to prevent arming with extremely bad/questionable servo power supply. The board rail was parameterized on the lower side as it allows the user to specify a higher minimum voltage then the default range, which can be convenient when dealing with backup regulators.

### CODE REVIEW

**Comment by proficnc** [comment]
I really like this feature! 
@bugobliterator? 
adding to this, there are status flags for checking that the two power rails are good on the cube... having these as part of the prearm would also be good

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
 #if HAL_HAVE_BOARD_VOLTAGE
     // check board voltage
     if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_VOLTAGE)) {
-        if(((hal.analogin->board_voltage() < AP_ARMING_BOARD_VOLTAGE_MIN) || (hal.analogin->board_voltage() > AP_ARMING_BOARD_VOLTAGE_MAX))) {
-            check_failed(ARMING_CHECK_VOLTAGE, report, "Check board voltage");
+        const float bus_voltage =  hal.analogin->board_voltage();
+        if(((bus_voltage < _vbus_min) || (hal.analogin->board_voltage() > AP_ARMING_BOARD_VOLTAGE_MAX))) {
```
Comment (peterbarker) [inline libraries/AP_Arming/AP_Arming.cpp]: Well, you *did* go to the trouble of pulling out a temporary variable.  Suggest reusing it here :-)

File: libraries/AP_Arming/AP_Arming.cpp [hunk 2]
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
...
+            check_failed(ARMING_CHECK_VOLTAGE, report, "Board voltage out of range (%1.1f)", bus_voltage);
```
Comment (peterbarker) [inline libraries/AP_Arming/AP_Arming.cpp]: could say, "%1.1f < %1.1f" here.

**Comment by WickedShell** [comment]
@peterbarker I struggled a bit with the status text clarity. I've tried to incorporate your idea here, but I presented it a bit differently. (I didn't like the `<` then in the board voltage check, because I would have needed 2 different check_failed's, or a more complicated string)

**Comment by WickedShell** [comment]
@Naterater as presented this won't yet help with in flight drop outs.

I'm wondering if the parameters should actually live in AP_BoardConfig and that way AP_Arming and AP_BattMonitor can grab them out of the same place? (AP_BattMonitor currently drives the logging of the `POWR` message, and has the most appropriate failsafes to reuse for this)

File: libraries/AP_Arming/AP_Arming.cpp [hunk 3]
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
...
-            check_failed(ARMING_CHECK_VOLTAGE, report, "Check board voltage");
+        const float bus_voltage =  hal.analogin->board_voltage();
+        if(((bus_voltage < _vbus_min) || (bus_voltage > AP_ARMING_BOARD_VOLTAGE_MAX))) {
+            check_failed(ARMING_CHECK_VOLTAGE, report, "Board voltage (%1.1fv) out of range %1.1f-%1.1fv", bus_voltage, _vbus_min, AP_ARMING_BOARD_VOLTAGE_MAX);
             return false;
         }
     }
 #endif
+
+    if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_VOLTAGE)) {
```
Comment (OXINARF) [inline libraries/AP_Arming/AP_Arming.cpp]: We should probably had a new macro for this as the HAL_HAVE_BOARD_VOLTAGE above.

**Comment by WickedShell** [comment]
Moved parameters to AP_BoardConfig for future use cases, added a HAL_HAVE_SERVO_VOLTAGE define to make the servo rail check optional.

Most importantly while doing this it was revealed that ChibiOS had disabled the board voltage arming checks due to never having set the define appropriately to allow the check to run. @tridge I'm adding it to the missing chibios list, but once we pull this in it can come right back off.

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 4]
```diff
@@ -773,6 +773,7 @@ def write_UART_config(f):
             '#define HAL_UART_IO_DRIVER ChibiOS::UARTDriver uart_io(HAL_UART_IOMCU_IDX)\n'
         )
         uart_list.append(config['IOMCU_UART'][0])
+        f.write('#define HAL_HAVE_SERVO_VOLTAGE 1\n') # make the assumption that IO gurantees servo monitoring
```
Comment (OXINARF) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py]: Nitpicking: guarantees

**Comment by OXINARF** [comment]
@davidbuzz Closing doesn't get it merged 😄 

@WickedShell There was a spelling error, but we chose to merge it anyway! Thanks!

**Pull request merged**